src/models/contact.ts

```diff
@@ -24,9 +38,23 @@ EmailAddress.attributes = {
   }),
 };
 
-class IMAddress extends Model {
-  type?: string;
-  imAddress?: string;
+// eslint-disable-next-line @typescript-eslint/interface-name-prefix
+export interface IMAddressProperties {
+  type: string;
+  imAddress: string;
+}
+
+class IMAddress extends Model implements IMAddressProperties {
+  type!: string;
+  imAddress!: string;
+
+  constructor(props?: IMAddressProperties) {
+    super(props);
+    if (!props) {
```

Comment:
Can we just initialize the props with `''` instead of having this check?

Reply:
Unfortunately, if we initialize props with anything other than an object of type `IMAddressProperties` (object with type and imAddress set) because of the strictness on those properties. The only way to do it like you're suggesting is to do something like `constructor(props: IMAddressProperties = {type: '', imAddress: ''}) {`

Alternatively, we can make all of these interfaces less strict allowing users to pass in as many or as little initial properties and in the do something like `this.type = props?.type ? props.type : "";` for every required property.

Reply:
I have re-worked the props to match what we talked about earlier. The properties now have default values without any extra checks.